site/source/_posts/2017/11/17-Dojo-2-Beta-4/Dojo-2-Beta-4.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Dojo 2 Beta 4
+date: 2017-11-17 18:00:00
+authorId: kitsonk
+featured_image: featured.png
+---
+
+The Beta 4 Release of Dojo 2 puts us on a clear path to releasing Dojo 2.  It has several significant changes and updates.
+
+{% asset_img featured.png feature-full %}
+
+<!-- more -->
+
+## Getting Beta 4
+
+Beta 4 is currently tagged as `latest` in npm, so all just updating to the latest version will provide you with Beta 4.  We also have a release tag of `beta4` in npm if you want to focus just on using Beta 4.
+
+With Beta 3, we revised our package versions to make it easier to keep a compatible set of packes.  We are releasing all the pre-release versions of Dojo 2 packages under major version `0`, but are introducing breaking changes in the minor release (e.g. `0.2.0` has breaking changes from `0.1.0`).  The versions we publish to npm rely upon a minor range.  You will see that some packages for Beta 4 are `0.3.#` and some are `0.2.#`, but they all depend on a range of version they know they are compatible with.  This means you don't have to rely upon a release tag.  So ideally you would want to ensure if you are managing your dependencies directly, you depend upon a minor release (e.g. `~0.2.0`) for packages.
+
+As we make releases of packages in anticipation of Beta 5, we will not publish those packages to `latest` until we release Beta 5.
+
```

Comment:
At this point, let's add a few brief paragraphs reintroducing the various packages, and then a paragraph or two on widget-core and widgets overall.

Reply:
I am not sure what you mean here.  I am not sure what value listing packages does.  If we want to do something like "Just getting started" and pointing them at a tutorial or some other location.  I will add something like that.

Reply:
I guess what I think is missing is something that says what the overall state of Dojo 2 beta 4 is, we just jump into the details. One way to do that is to quickly cover the major things Dojo 2 currently has and the state they are in, before diving in. Perhaps there's a better way, but what I'm asking for is a quick high level summary of where things are, what is stable and available, what remains to be done before Dojo 2 final, etc.